packages/openneuro-app/src/scripts/users/user-notification-accordion-actions.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useCallback } from "react"
+import * as Sentry from "@sentry/react"
+import { toast } from "react-toastify"
+import { useMutation } from "@apollo/client"
+import type { MappedNotification } from "../types/user-types"
+import { UPDATE_NOTIFICATION_STATUS_MUTATION } from "../queries/datasetEvents"
+import { StatusActionButton } from "./components/status-action-buttons"
+import iconUnread from "../../assets/icon-unread.png"
+import iconSaved from "../../assets/icon-saved.png"
+import iconArchived from "../../assets/icon-archived.png"
+import styles from "./scss/usernotifications.module.scss"
+import ToastContent from "../common/partials/toast-content.jsx"
+
+interface NotificationActionButtonsProps {
+  notification: MappedNotification
+  isProcessing: boolean
+  onUpdate: (id: string, updates: Partial<MappedNotification>) => void
+  setError: (error: string | null) => void
+  handleProcessAction: (action: "accepted" | "denied") => void
+}
+
+export const NotificationActionButtons: React.FC<
+  NotificationActionButtonsProps
+> = ({
+  notification,
+  isProcessing,
+  onUpdate,
+  setError,
+  handleProcessAction,
+}) => {
+  const {
+    id,
+    status,
+    type,
+    approval,
+  } = notification
+
+  const isContributorRequest = type === "approval"
+
+  const [updateNotificationStatus] = useMutation(
+    UPDATE_NOTIFICATION_STATUS_MUTATION,
+  )
+
+  // TODO just show errors for now
+  const handleStatusChange = useCallback(async (
+    newStatus: "unread" | "saved" | "archived",
+  ) => {
+    if (isProcessing) return
+
+    try {
+      const backendStatus = newStatus.toUpperCase()
+
+      await updateNotificationStatus({
+        variables: {
+          datasetEventId: id,
+          status: backendStatus,
+        },
+      })
```

Comment:
Ran into 400 error, Cannot query field "updateNotificationStatus" on type "Mutation". on triggering this.

Reply:
yes, I put that on hold, and have a question about how to store that data on a user if many users will share the same notification.